### ISSUE TITLE
Citation in tables/figures do not comply to specific APA style rule

### ISSUE THREAD

**Issue by mgkurtz** [plk/biblatex-apa#248]
Section 8.17 of the [APA style](https://openlibrary.org/books/OL28021359M/Publication_Manual_of_the_American_Psychological_Association) requires

> In tables and figures, use an ampersand between names in both parenthetical and narrative citations.

(I guess this excludes captions, at least this seems sensible and the example copyright attribution of Table 7.14 of the APA style manual spells out “[…], and M. Schaller”, likewise the copyright attribution in Figure 7.3.)

To comply with this rule, I imagine a new `\tablecite` command and have `\textcite` use it automatically if inside a table or figure.

For `\tablecite`, I copy pasted and edited `\parencite` to come up with this:

```tex
\DeclareDelimFormat[tablecite]{finalnamedelim}
  {\ifnum\value{liststop}>2 \finalandcomma\fi\addspace\&\space}
\DeclareCiteCommand{\tablecite}
  {\usebibmacro{cite:init}%
   \usebibmacro{prenote}}
  {\usebibmacro{citeindex}%
   \usebibmacro{cite}%
   \usebibmacro{cite:post}}
  {}
  {\usebibmacro{postnote}}
\DeclareMultiCiteCommand{\tablecites}{\tablecite}
  {\setunit{\multicitedelim}}
\DeclareCiteCommand*{\tablecite}[\mkbibparens]
  {\usebibmacro{cite:init}%
   \usebibmacro{prenote}}
  {\usebibmacro{citeindex}%
   \usebibmacro{citeyear}%
   \usebibmacro{cite:post}}
  {}
  {\usebibmacro{postnote}}
```

Concerning recognition of tables and figures, setting a switch in a local scope should work:

```tex
\newif\ifintable
\AddToHook{env/table/begin}{\intabletrue}
\AddToHook{env/figure/begin}{\intabletrue}
\AddToHook{cmd/@makecaption/before}{\intablefalse}
```

I also asked on [TeX Stack Exchange](https://tex.stackexchange.com/questions/729733/how-to-determine-reliably-whether-code-is-called-in-a-table-or-figure-environmen) to get feedback whether this approach works in setups using various kinds of packages.

**Comment by mgkurtz** [comment]
> Table 8.1 on p. 266 of APA 7 says ampersand between two names in parenthetical and normal "and" in narrative.

That is right. However, two paragraphs below that table, the APA style says (as I quoted above) that citations in tables and figures are to use ampersands in narrative citations as well.

According to that rule, Table 8.1 and possibly some figures would of course have to use ampersands throughout. But that would of course defeat the purpose of showing off their other rules in tables and figures. Thus, it seems to me, that for demonstration purposes they deliberately ignored their own rule here.

**Comment by plk** [comment]
try `apa.cbx` from the master branch - I have used your basic table detection for now.

**Comment by mgkurtz** [comment]
Thanks a lot 🙏

I would maybe add documentation for the `\intabletrue`/`\intablefalse` switches. So, that people can set or overwrite them, if wanted.

**Comment by plk** [comment]
I think I'll leave it for now in case we get a better solution using Lua - I'll release a new version shortly.
